src/Nethermind/Nethermind.BeaconNode/AttestationProducer.cs

```diff
@@ -0,0 +1,126 @@
+            bool proofOfCustodyBit, Slot slot, CommitteeIndex index,
+            CancellationToken cancellationToken)
+        {
+            Root head = await _store.GetHeadAsync().ConfigureAwait(false);
+            BeaconBlock headBlock = (await _store.GetSignedBlockAsync(head).ConfigureAwait(false)).Message;
+            BeaconState parentState = await _store.GetBlockStateAsync(head).ConfigureAwait(false);
+
+            // Clone state (will mutate) and process outstanding slots
+            BeaconState headState = BeaconState.Clone(parentState);
+            _beaconStateTransition.ProcessSlots(headState, slot);
+
+            // Set attestation_data.index = index where index is the index associated with the validator's committee.
+            ValidatorIndex? validatorIndex = FindValidatorIndexByPublicKey(headState, validatorPublicKey);
+
+            if (validatorIndex == null)
+            {
+                throw new Exception($"Can not find validator index for public key {validatorPublicKey}");
+            }
+
+            // TODO: May need a more efficient way to try and find the committee and position within the committee.
+            // Some of this may already be cached in Validator Assignments (generally stable for an epoch),
+            // but not with the index within the committee. Easy enough to extend and use the same cache.
+
+            IReadOnlyList<ValidatorIndex> committee =
+                _beaconStateAccessor.GetBeaconCommittee(headState, headState.Slot, index);
+            int committeeSize = committee.Count;
+
+            int? committeeMemberIndexOfValidator = null;
+            for (int committeeMemberIndex = 0; committeeMemberIndex < committee.Count; committeeMemberIndex++)
+            {
+                if (committee[committeeMemberIndex] == validatorIndex)
+                {
+                    committeeMemberIndexOfValidator = committeeMemberIndex;
+                    break;
+                }
+            }
+
+            if (committeeMemberIndexOfValidator == null)
+            {
+                throw new Exception($"Validator index {validatorIndex} is not a member of committee {index}");
+            }
+
+            Root beaconBlockRoot = _cryptographyService.HashTreeRoot(headBlock);
+
+            Checkpoint source = headState.CurrentJustifiedCheckpoint;
+
+            Epoch currentEpoch = _beaconStateAccessor.GetCurrentEpoch(headState);
+            Slot startSlot = _beaconChainUtility.ComputeStartSlotOfEpoch(currentEpoch);
+            Root epochBoundaryBlockRoot;
+            if (startSlot == headState.Slot)
+            {
+                epochBoundaryBlockRoot = beaconBlockRoot;
+            }
+            else
+            {
+                epochBoundaryBlockRoot = _beaconStateAccessor.GetBlockRootAtSlot(headState, startSlot);
+            }
+
+            Checkpoint target = new Checkpoint(currentEpoch, epochBoundaryBlockRoot);
```

Comment:
var, you use it later in same situation so lets be consistent

Reply:
not important enough to chase in code review?